<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body></body>
  <script>
    const lodscript = async (src) => {
      return new Promise((resolve, reject) => {
        let script = document.createElement("script");
        script.src = src;
        document.head.append(script);
        script.onload = () => {
          resolve(src + " Done susces");
        };
      });
    };
    //  1>
    let a = lodscript(
      "https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/js/bootstrap.bundle.min.js"
    );

    a.then((value) => {
      console.log(value);
    });

    2
    const main2 = async()=>{
        let a= await lodscript("https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/js/bootstrap.bundle.min.js")

        console.log(a)
        console.log(new Date().getMilliseconds()) // show the time

    }
    main2();

    3>

    let p = () => {
      new Promise((resolve, reject) => {
        setTimeout(() => {
          reject(new Error("plese this is not acteble"));
        }, 3000);
      });
    }

    let z= async()=>{

        try{
            let c= await p()  // try cathch handele
            console.log(c)
        }
        catch(err){
            console.log("this error can be handeled")
        }
           
    }
    z();


    let p1= async()=>{
        return new Promise((resolve, reject) => {
            setTimeout(() => {
                resolve(359)
            }, 1000);
        })
    }

    let p2= async()=>{
        return new Promise((resolve, reject) => {
            setTimeout(() => {
                resolve(30)
            }, 2000);
        })
    }
    let p3= async()=>{
        return new Promise((resolve, reject) => {
            setTimeout(() => {
                resolve(40)
            }, 3000);
        })
    }
  </script>
</html>
